src/Reprocessing.re

```diff
@@ -120,177 +124,193 @@ let run =
-      let textureBuffer = Gl.createTexture(~context=env.gl);
-      Gl.bindTexture(~context=env.gl, ~target=RGLConstants.texture_2d, ~texture=textureBuffer);
-      Gl.texImage2D_RGBA(
-        ~context=env.gl,
-        ~target=RGLConstants.texture_2d,
-        ~level=0,
-        ~width,
-        ~height,
-        ~border=0,
-        ~data
-      );
-      Gl.texParameteri(
-        ~context=env.gl,
-        ~target=RGLConstants.texture_2d,
-        ~pname=RGLConstants.texture_mag_filter,
-        ~param=RGLConstants.linear
-      );
-      Gl.texParameteri(
-        ~context=env.gl,
-        ~target=RGLConstants.texture_2d,
-        ~pname=RGLConstants.texture_min_filter,
-        ~param=RGLConstants.linear
-      );
-      Gl.texParameteri(
-        ~context=env.gl,
-        ~target=RGLConstants.texture_2d,
-        ~pname=RGLConstants.texture_wrap_s,
-        ~param=RGLConstants.clamp_to_edge
-      );
-      Gl.texParameteri(
-        ~context=env.gl,
-        ~target=RGLConstants.texture_2d,
-        ~pname=RGLConstants.texture_wrap_t,
-        ~param=RGLConstants.clamp_to_edge
-      );
-      () => {
-        let (x, y) = (0, 0);
-        let (x1, y1) = (float_of_int @@ x + width, float_of_int @@ y);
-        let (x2, y2) = (float_of_int(x), float_of_int @@ y);
-        let (x3, y3) = (float_of_int @@ x + width, float_of_int @@ y + height);
-        let (x4, y4) = (float_of_int(x), float_of_int @@ y + height);
-        let verticesColorAndTexture = [|
-          x1,
-          y1,
-          0.0,
-          0.0,
-          0.0,
-          0.0,
-          1.,
-          1.0,
-          1.0,
-          x2,
-          y2,
-          0.0,
-          0.0,
-          0.0,
-          0.0,
-          1.,
-          0.0,
-          1.0,
-          x3,
-          y3,
-          0.0,
-          0.0,
-          0.0,
-          0.0,
-          1.,
-          1.0,
-          0.0,
-          x4,
-          y4,
-          0.0,
-          0.0,
-          0.0,
-          0.0,
-          1.,
-          0.0,
-          0.0
-        |];
-        drawGeometry(
-          ~vertexArray=Gl.Bigarray.of_array(Gl.Bigarray.Float32, verticesColorAndTexture),
-          ~elementArray=Gl.Bigarray.of_array(Gl.Bigarray.Uint16, [|0, 1, 2, 1, 2, 3|]),
-          ~mode=RGLConstants.triangles,
-          ~count=6,
-          ~textureBuffer,
-          env
+        /*** Start the render loop. **/
+        Gl.render(
+          ~window=env.window,
+          ~displayFunc=
+            (f) => {
+              if (env.frame.count === 2) {
+                reDrawPreviousBufferOnSecondFrame()
+              };
+              switch hotreloadData^ {
+              | Some(_) => ignore @@ Reprocessing_Hotreload.checkRebuild(fns.filename)
+              | None => ()
+              };
+              let beforeDrawTime = Gl.getTimeMs();
+              userState := fns.draw(userState^, env);
+              let afterDrawTime = Gl.getTimeMs();
```

Comment:
I added this timing stuff and the text drawing for performance monitoring.
(end of changes to this file)